### DISCUSSION TITLE
getting fock states directly for use

### DISCUSSION THREAD

**Discussion by sls26** [QuSpin/QuSpin#649] [Tips and Tricks with QuSpin]
Since Quspin is already calculating the basis, is there any way to access it directly?

for example, for a 3-particle boson system,it is being printed out as 

reference states: 
array index   /   Fock state   /   integer repr. 
      0.         |3 0 0>           48  
      1.         |2 1 0>           36  
      2.         |2 0 1>           33  
      3.         |1 2 0>           24  
      4.         |1 1 1>           21  
      5.         |1 0 2>           18  
      6.         |0 3 0>           12  
      7.         |0 2 1>            9  
      8.         |0 1 2>            6  
      9.         |0 0 3>            3  
      
where I only need fock states.

**Comment by mgbukov** [comment]
we don't store the basis vectors in quspin; instead, you can access the
integer representations using basis.states

To convert between the integer and Flock representations, see this tutorial:

https://quspin.github.io/QuSpin/examples/example00.html#example-00-label

On Tue, Nov 7, 2023 at 6:48 AM sls26 ***@***.***> wrote:

> Since Quspin is already calculating the basis, is there any way to access
> it directly?
>
> for example, for a 3-particle boson system,it is being printed out as
>
> reference states:
> array index / Fock state / integer repr.
> 0. |3 0 0> 48
> 1. |2 1 0> 36
> 2. |2 0 1> 33
> 3. |1 2 0> 24
> 4. |1 1 1> 21
> 5. |1 0 2> 18
> 6. |0 3 0> 12
> 7. |0 2 1> 9
> 8. |0 1 2> 6
> 9. |0 0 3> 3
>
> where I only need fock states.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/QuSpin/QuSpin/discussions/649>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADHSUBW6YKQWDHIARFHUDJ3YDHDR7AVCNFSM6AAAAAA7ATQGISVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZVHAZDCMZUHA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Marin Bukov, PhD
Research Group Leader
Max Planck Institute for the Physics of Complex Systems
Noethnitzer Str. 38
Dresden 01187, Germany

url: https://www.pks.mpg.de/nqd
<http://quantum-dynamics.phys.uni-sofia.bg/>
